### PR TITLE
Update the Google builder in the list of known builders.

### LINKED ISSUE #2199: Update Google's suggested builder
### Description
Currently the Pack CLI's suggested builders list contains an outdated Google builder image, `gcr.io/buildpacks/builder:v1`, which is based on the EOL Ubuntu 18.04 LTS:
https://github.com/buildpacks/pack/blob/ce8db3c5d11b8e040fd565632668a697e53e4f87/internal/builder/known_builder.go#L12-L18

### Proposed solution
- Add one of the newer Google builder images documented here: https://cloud.google.com/docs/buildpacks/builders
- Then either remove `gcr.io/buildpacks/builder:v1` from the list entirely, or demote it to only being a "trusted" builder and not also a "suggested" builder (now that the concepts are separate, after https://github.com/buildpacks/pack/pull/2043)

### Describe alternatives you've considered
Continue to suggest a builder image that uses an EOL and insecure Ubuntu version.

### PR DESCRIPTION
**Pull request by hhiroshell**
## Summary
This PR updates the version of the [Google builder](https://cloud.google.com/docs/buildpacks/builders) in the KnownBuilders addressing #2199 

In this PR, I have removed `gcr.io/buildpacks/builder:v1` from both `Trusted` and `Suggested`. If there are criteria for being `Trusted` or `Suggested` that require keeping v1, please let me know. I will be happy to make the necessary adjustments.

#### Before

- `pack builder suggest`

```
$ pack builder suggest
Suggested builders:
        Google:                gcr.io/buildpacks/builder:v1                            GCP Builder for all runtimes
        Heroku:                heroku/builder:22                                       Heroku-22 (Ubuntu 22.04) base image with buildpacks for Go, Java, Node.js, PHP, Python, Ruby & Scala
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-base                     Small base image with buildpacks for Java, Node.js, Golang, .NET Core, Python & Ruby
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-buildpackless-static     Static base image (jammy build image, distroless run image) suitable for static binaries like Go or Rust
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-full                     Larger base image with buildpacks for Java, Node.js, Golang, .NET Core, Python, Ruby, & PHP
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-tiny                     Tiny base image (jammy build image, distroless run image) with buildpacks for Golang & Java
```

- `pack config trusted-builders list`

```
$ pack config trusted-builders list
Trusted Builders:
  gcr.io/buildpacks/builder:v1
  heroku/builder:20
  heroku/builder:22
  paketobuildpacks/builder-jammy-base
  paketobuildpacks/builder-jammy-buildpackless-static
  paketobuildpacks/builder-jammy-full
  paketobuildpacks/builder-jammy-tiny
```

#### After

- `pack builder suggest`

```
$ pack builder suggest
Suggested builders:
        Google:                gcr.io/buildpacks/builder:google-22                     GCP Builder for all runtimes
        Heroku:                heroku/builder:24                                       Ubuntu 24.04 AMD64+ARM64 base image with buildpacks for Go, Java, Node.js, PHP, Python, Ruby & Scala.
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-base                     Small base image with buildpacks for Java, Node.js, Golang, .NET Core, Python & Ruby
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-buildpackless-static     Static base image (jammy build image, distroless run image) suitable for static binaries like Go or Rust
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-full                     Larger base image with buildpacks for Java, Node.js, Golang, .NET Core, Python, Ruby, & PHP
        Paketo Buildpacks:     paketobuildpacks/builder-jammy-tiny                     Tiny base image (jammy build image, distroless run image) with buildpacks for Golang & Java
```

- `pack config trusted-builders list`

```
$ pack config trusted-builders list
Trusted Builders:
  gcr.io/buildpacks/builder:google-22
  heroku/builder:20
  heroku/builder:22
  heroku/builder:24
  paketobuildpacks/builder-jammy-base
  paketobuildpacks/builder-jammy-buildpackless-static
  paketobuildpacks/builder-jammy-full
  paketobuildpacks/builder-jammy-tiny
```

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

Resolves #2199

### CODE REVIEW

File: internal/builder/known_builder.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ type KnownBuilder struct {
 var KnownBuilders = []KnownBuilder{
 	{
 		Vendor:             "Google",
-		Image:              "gcr.io/buildpacks/builder:v1",
+		Image:              "gcr.io/buildpacks/builder:google-22",
 		DefaultDescription: "GCP Builder for all runtimes",
```
Comment (edmorley) [inline internal/builder/known_builder.go:15]: I think we should also sync the description here with the description in the builders upstream manifest:
https://github.com/GoogleCloudPlatform/buildpacks/blob/ccee07f7459d6e1a5efe15258f99d7c09e2fcae5/builders/gcp/base/google.22.builder.toml#L1

```suggestion
		DefaultDescription: "Ubuntu 22.04 base image with buildpacks for .NET, Dart, Go, Java, Node.js, PHP, Python, and Ruby",
```

cc @matthewrobertson and @kritkasahni-google as recent upstream builder committers:
https://github.com/GoogleCloudPlatform/buildpacks/commits/main/builders/gcp/base/google.22.builder.toml

Comment (hhiroshell) [reply internal/builder/known_builder.go:15]: Thank you for pointing it out; it looks fine to me as well. 😄

**Comment by jjbustamante** [comment]
Do we want to keep older builders as *trusted* but not *suggested* as we do with Heroku? 

@natalieparellano , @edmorley

**Review by jjbustamante — APPROVED** [review]

**Comment by edmorley** [comment]
> Do we want to keep older builders as _trusted_ but not _suggested_ as we do with Heroku?

For the Heroku builders, we've used the following methodology:
- remove any builders that are EOL (eg those based on Ubuntu 18.04)
- have all other active Heroku builders be trusted
- have only the most recent Heroku builder (that based on Ubuntu 24.04) be suggested

Given that the `gcr.io/buildpacks/builder:v1` builder is based on Ubuntu 18.04 and so EOL, I think it makes sense to remove it entirely from trusted builders.

**Pull request merged**